src/codegeneration/InstantiateModuleTransformer.js

```diff
@@ -82,7 +83,15 @@ class DeclarationExtractionTransformer extends HoistVariablesTransformer {
     this.declarations_ = [];
   }
   getDeclarationStatements() {
-    return [this.getVariableStatement(), ...this.declarations_];
+    let statement = this.getVariableStatement();
+    let statements;
+    if (statement.type === ANON_BLOCK) {
```

Comment:
The reason we get into this state is that `getVariableStatement` sometimes returns an `AnonBlock` and anon blocks are not allowed in the final tree.

Yeah, this is kind of ugly. I'm tempted to create a function called `flattenAnonBlocks(statements)` that spreads the anon blocks like we do in transformList.